LSPS2/README.md

```diff
@@ -0,0 +1,736 @@
+
+* The client node ID (or any public key whose private key the
+  client knows).
+* The `jit_channel_scid`.
+* The LSP node ID.
+* The `lsp_cltv_expiry_delta`.
+* The expiry for the payment.
+* Whether it can use MPP or not.
+
+However, the client MAY provide this information to the payer
+by means other than BOLT11 invoice, including in a form that
+is not decodable by the payer, but which can be decoded by
+another LN participant.
+
+The only necessary requirement is for the LSP to see an incoming
+HTLC onion that indicates the `jit_channel_scid` as the next hop,
+and as long as the client is able to cause the payer to do this,
+it is would be compatible with this protocol.
+
+> **Non-normative** For example, the client node ID, the
+> `jit_channel_scid`, the LSP node ID, and the
+> `lsp_cltv_expiry_delta` can be encoded as the last hop of
+> a blinded path, for example in a BOLT12 offer or invoice.
+> The payer would not have direct access to the information,
+> but the introduction point of the blinded path would be able
+> to start the decoding of the blinded path.
+
+### 4.  Payment
+
+The client MUST issue the generated invoice to the payer of the
+client, or equivalently inform the payer of the necessary
+information.
+
+The payer can then generate an outgoing payment part to
+deliver the paid amount to the LSP.
+If the client selected "MPP+fixed-invoice" mode, the payer can
+generate multiple outgoing payments parts.
+
+In "MPP+fixed-invoice" mode, the LSP, if it receives a forward
+where the next hop is `jit_channel_scid`:
+
+* MUST wait for payment parts of a single payment hash until all
+  parts sum up to at least `payment_size`.
+* MUST NOT fail with `mpp_timeout`, as the final hop is the client
+  and that error is only allowed at the final hop.
+* MUST fail with `unknown_next_peer` if all parts have **NOT**
+  summed up to at least `payment_size` **AND**
+  `opening_fee_params.valid_until` has passed.
+* MAY fail with `unknown_next_peer` if it receives too many payment
+  parts.
+* When all conditions below are true, MUST continue to the next
+  step:
+  * The current time is strictly less than
+    `opening_fee_params.valid_until`.
+  * All received payment parts sum up to at least `payment_size`
+    plus `opening_fee`.
+  * The client is connected.
+
+In "no-MPP+var-invoice" mode, the LSP, if it receives a forward
+where the next hop is `jit_channel_scid`, before
+`opening_fee_params.valid_until`:
+
+* MUST set `payment_size` as the forwarded value of the HTLC.
+* MUST compute the `opening_fee` based on the `payment_size`, and
+  if the computation overflows, MUST fail with `unknown_next_peer`.
+* MUST check that `opening_fee + htlc_minimum_msat < payment_size`,
+  and if that fails, MUST fail with `unknown_next_peer`.
+
+### 5.  Channel Opening And Forwarding
+
+The LSP requests a channel open to the client.
+
+The LSP selects the channel size.
+The LSP SHOULD scale the size of the channel based on the
+`payment_size`.
```

Comment:
I think this line can be removed. Receiving a large payment is not a guarantee that the client will receive large payments again later. It's either up to the LSP to decide the channel size or up to the client and it can buy the channel size it wants.

Reply:
Done.